sys/net/gnrc/netif/lorawan/gnrc_netif_lorawan.c

```diff
@@ -351,6 +351,11 @@ static int _set(gnrc_netif_t *netif, const gnrc_netapi_opt_t *opt)
     switch (opt->opt) {
         case NETOPT_LORAWAN_DR:
             assert(opt->data_len == sizeof(uint8_t));
+            if (*((uint8_t *)opt->data) & 0xF0) {
+                DEBUG("gnrc_netif: Invalid datarate\n")
+                gnrc_netif_release(netif);
+                return -EINVAL;
```

Comment:
instead of returning here, please set the return value (`res`) to -EINVAL and don't set the datarate. Otherwise we have more than one exit point for this function

Reply:
Thanks for the suggestion :)